src/test/java/io/jenkins/plugins/coverage/model/CoveragePluginITest.java

```diff
@@ -0,0 +1,264 @@
+        assertThat(coverageResult.getLineCoverage()).isEqualTo(new Coverage(6083, 6368-6083));
+    }
+
+    @Test
+    public void CoveragePluginPipeline_2_JacocoInputFile(){
+        WorkflowJob job = createPipelineWithWorkspaceFiles(JACOCO_FILE_NAME, JACOCO_LESS_FILE_NAME);
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [jacocoAdapter('**/*.xml')]"
+                + "}", true));
+
+        Run<?, ?> build = buildSuccessfully(job);
+        assertThat(build.getNumber()).isEqualTo(1);
+
+        CoverageBuildAction coverageResult = build.getAction(CoverageBuildAction.class);
+        assertThat(coverageResult.getLineCoverage()).isEqualTo(new Coverage(12166, 12736-12166));
+    }
+
+
+
+    @Test
+    public void CoveragePluginPipeline_0_CoberturaInputFile(){
+        WorkflowJob job = createPipelineWithWorkspaceFiles();
+
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [cobertura('cobertura.xml')]"
+                + "}", true));
+
+        Run<?, ?> build = buildSuccessfully(job);
+        assertThat(build.getNumber()).isEqualTo(1);
+
+        CoverageBuildAction coverageResult = build.getAction(CoverageBuildAction.class);
+        assertThat(coverageResult).isEqualTo(null);
+    }
+
+    @Test
+    public void CoveragePluginPipeline_1_CoberturaInputFile(){
+        WorkflowJob job = createPipelineWithWorkspaceFiles(COBERTURA_FILE_NAME);
+
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [cobertura('cobertura.xml')]"
+                + "}", true));
+
+        Run<?, ?> build = buildSuccessfully(job);
+        assertThat(build.getNumber()).isEqualTo(1);
+
+        CoverageBuildAction coverageResult = build.getAction(CoverageBuildAction.class);
+        assertThat(coverageResult.getLineCoverage()).isEqualTo(new Coverage(2, 0));
+    }
+
+    @Test
+    public void CoveragePluginPipeline_2_CoberturaInputFile(){
+        WorkflowJob job = createPipelineWithWorkspaceFiles(COBERTURA_FILE_NAME, COBERTURA_FILE_NAME_2);
+
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [cobertura('**/*.xml')]"
+                + "}", true));
+
+        Run<?, ?> build = buildSuccessfully(job);
+        assertThat(build.getNumber()).isEqualTo(1);
+
+        CoverageBuildAction coverageResult = build.getAction(CoverageBuildAction.class);
+        assertThat(coverageResult.getLineCoverage()).isEqualTo(new Coverage(4, 0));
+    }
+
+
+    @Test
+    public void CoveragePluginPipeline_1_CoberturaInputFile_1JacocoInputFile() throws IOException {
+        WorkflowJob job = createPipelineWithWorkspaceFiles(JACOCO_FILE_NAME, COBERTURA_FILE_NAME);
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "publishCoverage adapters: [cobertura('**/*.xml'),jacocoAdapter('**/*.xml')]"
```

Comment:
This makes no sense, each adapter should read only files that it can understand.